src/WebJobs.Script/Config/ScriptHostRecycleOptions.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Azure.WebJobs.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    /// <summary>
+    /// Options that control Script Host recycling behavior.
+    /// </summary>
+    public sealed class ScriptHostRecycleOptions : IOptionsFormatter
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether sequential host restarts are required.
+        /// </summary>
+        public bool SequentialHostRestartRequired { get; set; }
+
+        public string Format()
+        {
+            return JsonSerializer.Serialize(this, typeof(ScriptHostRecycleOptions), ScriptHostRecycleOptionsJsonContext.Default);
+        }
+    }
+
+    [JsonSourceGenerationOptions(WriteIndented = true)]
```

Comment:
Consider using serialization optimized mode:
https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation#serialization-optimization-fast-path-mode-example


```suggestion
    [JsonSourceGenerationOptions(
        GenerationMode = JsonSourceGenerationMode.Serialization, WriteIndented = true)]
```